homeassistant/components/automation/zone.py

```diff
@@ -14,23 +14,31 @@
 from homeassistant.helpers import (
     condition, config_validation as cv, location)
 
+CONF_ENTITY_ID_PATTERN = 'entity_id_pattern'
```

Comment:
I'm not 100% sure yet if a pattern is the right solution here, it's making things complicated, especially for the UI. I wonder if a new `geo_location` automation trigger would be better.

Reply:
Sure, let me look into making this a separate trigger. I would still allow to define a pattern though. And I could look at solving the `leave` event at the same time.